example/simple_compute.cpp

```diff
@@ -0,0 +1,302 @@
+#include <VkBootstrap.h>
+#include <iostream>
+#include <fstream>
+#include "example_config.h"
+
+struct Init {
+    vkb::Instance instance;
+    vkb::InstanceDispatchTable inst_disp;
+    vkb::Device device;
+    vkb::DispatchTable disp;
+};
+
+struct Data {
+    VkQueue queue;
+
+    VkPipelineLayout pipeline_layout;
+    VkPipeline compute_pipeline;
+
+    VkCommandPool command_pool;
+    VkCommandBuffer command_buffer;
+
+    VkBuffer buffer_a;
+    VkDeviceMemory memory_a;
+    VkBuffer buffer_b;
+    VkDeviceMemory memory_b;
+
+    VkDescriptorSet descriptor_set;
+    VkDescriptorPool descriptor_pool;
+    VkDescriptorSetLayout descriptor_set_layout;
+};
+
+int device_initialization(Init& init) {
+    vkb::InstanceBuilder instance_builder;
+    auto instance_ret = instance_builder.use_default_debug_messenger().request_validation_layers().build();
+    if (!instance_ret) {
+        std::cout << instance_ret.error().message() << "\n";
+        return -1;
+    }
+    init.instance = instance_ret.value();
+
+    init.inst_disp = init.instance.make_table();
+
+    vkb::PhysicalDeviceSelector phys_device_selector(init.instance);
+    // Skip vk-bootstrap trying to create WSI for you
+    auto phys_device_ret = phys_device_selector.defer_surface_initialization().select();
```

Comment:
Defer surface initialization isnt for headless, rather its to make vk-bootstrap not do any checking of surface & physical device compatibility. Aka remove it pls thx.